atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/CqlClientImpl.java

```diff
@@ -0,0 +1,336 @@
+import com.datastax.driver.core.QueryOptions;
+import com.datastax.driver.core.RemoteEndpointAwareJdkSSLOptions;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.ThreadingOptions;
+import com.datastax.driver.core.exceptions.NoHostAvailableException;
+import com.datastax.driver.core.policies.AddressTranslator;
+import com.datastax.driver.core.policies.DefaultRetryPolicy;
+import com.datastax.driver.core.policies.LatencyAwarePolicy;
+import com.datastax.driver.core.policies.LoadBalancingPolicy;
+import com.datastax.driver.core.policies.RoundRobinPolicy;
+import com.datastax.driver.core.policies.TokenAwarePolicy;
+import com.datastax.driver.core.policies.WhiteListPolicy;
+import com.google.common.util.concurrent.AsyncFunction;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfig;
+import com.palantir.common.base.Throwables;
+import com.palantir.common.concurrent.NamedThreadFactory;
+import com.palantir.common.concurrent.PTExecutors;
+import com.palantir.conjure.java.config.ssl.SslSocketFactories;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.tracing.AsyncTracer;
+
+public final class CqlClientImpl implements CqlClient {
+
+    static class SimpleAddressTranslator implements AddressTranslator {
+
+        private final Map<String, InetSocketAddress> mapper;
+
+        SimpleAddressTranslator(CassandraKeyValueServiceConfig config) {
+            this.mapper = config.addressTranslation();
+        }
+
+        @Override
+        public void init(Cluster cluster) {
+        }
+
+        @Override
+        public InetSocketAddress translate(InetSocketAddress address) {
+            InetSocketAddress temp = mapper.getOrDefault(address.getHostString(), address);
+            return new InetSocketAddress(temp.getAddress(), temp.getPort());
+        }
+
+        @Override
+        public void close() {
+        }
+    }
+
+    private static final AtomicLong cassandraId = new AtomicLong();
+    private static final Logger log = LoggerFactory.getLogger(CqlClientImpl.class);
+
+    private final String clientName;
+    private final Executor executor;
+
+    private final ScheduledExecutorService healthCheckExecutor = PTExecutors.newSingleThreadScheduledExecutor(
+            new NamedThreadFactory(sessionName() + "-healthCheck", true /* daemon */));
+    private final PreparedStatement healthCheckStatement;
+    private final Cluster cluster;
+    private final Session session;
+
+    public static CqlClient create(CassandraKeyValueServiceConfig config, Set<InetSocketAddress> servers) {
+        long curId = cassandraId.getAndIncrement();
+        String clusterName = "cassandra" + "-" + curId;
```

Comment:
might it make sense for the cluster name to be just the keyspace? could be wrong here, plus you needn't do this anyway, if this is unset, a clustername gets generated for you I believe.